scarb/src/core/registry/client/http.rs

```diff
@@ -146,12 +151,77 @@ impl<'c> RegistryClient for HttpRegistryClient<'c> {
     }
 
     async fn supports_publish(&self) -> Result<bool> {
-        // TODO(mkaput): Publishing to HTTP registries is not implemented yet.
-        Ok(false)
+        let index_config = self.index_config.load().await?;
+        match index_config.upload {
+            Some(_) => Ok(true),
+            None => Ok(false),
+        }
```

Comment:
```suggestion
        Ok(self.index_config.load().await?.is_some())
```

Reply:
Changed in https://github.com/software-mansion/scarb/pull/1570/commits/148f40b897c820a0b647814661325600e2b92de9